### PR TITLE
Create functioning game

### PR DESCRIPTION
**Pull request by KojinKuro**
# Description

Integrate everyone's code to make a functional game w/ game logic. This completes the MVP for Iteration 1. Includes code refactoring for style matching, keeping code DRY, and clarity.

## Context

There is now a functional game. All of the features from the MVP have been included. To see how it works, please read the code. A good place to start is the `<Game />` component.
Features:
- Will compare your guess with the answer and tell you how similar they are
- Incremental unblurring as you guess more
- Winner / loser display
- Keyboard controls
- Randomly selecting a game every refresh

Furthermore, I did one pass over the code to make it clean. If anyone wants to look over or do a pass of their own, make a branch off this to merge into this one.
Refactor changes:
- Reorganized files into an organized folder structure
- Changed how imports, class names, camelCase vs kebab-case, etc to all be consistent across the application
- Renamed variables to match how they are refereed to else where in the application for clarity and consistency
- Deleted unneeded NPM packages.

Final note: It has come to my attention that the `generateNumberFromSeed` function does not work. The `crypto` npm library is part of Node.js meaning that `generateNumberFromSeed` can only run on the server. This doesn't really matter because that was the intention to begin with, but is something to be aware of.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

The best way to test this code is by running the game in the browser using `npm run dev`. Before running please use `npm i` to clean the packages and match the packages that I am using.

Some notes about the game for anyone who is testing it:
- The number of guesses and game data can be change through the props in `<Game />`.
- You must type the correct title as a game in the `mockGames` file. It does not need to match case, but must be the same letters and spacing. If not an error will display on the screen saying your game could not be found.
- The game will only let you enter guesses until you win or run out of guesses.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by moth-dust — APPROVED** [review]
Looks good!

File: src/components/Game/Game.jsx
Comment (BrandonDoza) [inline src/components/Game/Game.jsx:1]: Very well put together!

**Review by BrandonDoza — COMMENTED** [review]
Well done!

**Comment by LISims88** [comment]
I enjoyed being able to follow the logic of the game with css. It makes for an interesting thing.

**Pull request merged**